geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java

```diff
@@ -89,12 +93,51 @@ public Object execute(Object command, ParseResult parseResult) {
     }
   }
 
-  protected Object invokeCommand(Object command, ParseResult parseResult) {
+  protected Object invokeCommand(Object command, GfshParseResult parseResult) {
     // if no command instance is passed in, use the one in the parseResult
     if (command == null) {
       command = parseResult.getInstance();
     }
-    return ReflectionUtils.invokeMethod(parseResult.getMethod(), command,
-        parseResult.getArguments());
+
+    Object result =
+        ReflectionUtils.invokeMethod(parseResult.getMethod(), command, parseResult.getArguments());
+
+    if (!(command instanceof SingleGfshCommand)) {
+      return result;
+    }
+
+    SingleGfshCommand gfshCommand = (SingleGfshCommand) command;
+    CommandResult commandResult = (CommandResult) result;
+    if (commandResult.getStatus() == Result.Status.ERROR) {
+      return result;
+    }
+
+    // if command result is ok, we will need to see if we need to update cluster configuration
+    ClusterConfigurationService ccService = gfshCommand.getConfigurationService();
+    if (parseResult.getParamValue("member") != null || ccService == null) {
```

Comment:
I love seeing things like "And remember not to persist of the command was given with `--member`" removed from the collective memory and put into implementation.